app/client/components/consent/OnOffRadio.tsx

```diff
@@ -1,86 +1,99 @@
+    display: inline-block;
+    width: ${size.small}px;
+    height: ${size.small}px;
+    margin: 0 ${space[1]}px 0 0;
+    border: 2px solid currentColor;
+    border-radius: 50%;
+    position: relative;
+    transition: box-shadow ${transitions.short};
+    transition-delay: 0.08s;
+    :focus {
+      ${focusHalo};
+    }
+    :after {
+      background: currentColor;
+      position: absolute;
+      content: "";
+      top: 0;
+      right: 0;
+      bottom: 0;
+      left: 0;
+      border-radius: 50%;
+      transform: scale(0);
+      transform-origin: center;
+      transition: transform ${transitions.short};
+    }
+    :checked {
+      color: ${palette.neutral[100]};
+      &:after {
+        transform: scale(0.6);
+      }
+    }
+    :disabled {
+      color: ${palette.brand.pastel};
+      &:checked {
+        color: ${palette.brand.pastel};
+      }
+      border-color: ${palette.brand.pastel};
+    }
+    :not([disabled]) {
+      cursor: pointer;
+    }
+  }
 `;
-
-/**
- * backface-visibility: hidden fixes blur of text
- * caused by transforming scale of :after
- */
-const radioLabelStyles = (isSelected: boolean) => css`
+const radioLabelStyles = (disabled: boolean) => css`
+  margin-left: ${space[2]}px;
+  cursor: ${disabled ? "default" : "pointer"};
   display: flex;
   align-items: center;
-  position: relative;
-  font-size: 16px;
-  line-height: 22px;
-  font-family: "Guardian Text Sans Web", Helvetica Neue, Helvetica, Arial,
-    Lucida Grande, sans-serif;
-  font-weight: 600;
-  backface-visibility: hidden;
-
-  :before {
-    display: inline-block;
-    content: "";
-    width: 18px;
-    height: 18px;
-    border: 1px solid
-      ${isSelected ? `${palette.blue.header}` : `${palette.neutral[5]}`};
-    box-shadow: inset 0 0 0 3px ${palette.white};
-    border-radius: 60px;
-    margin-left: 10px;
-    margin-right: 5px;
-    transition: background 0.2s, box-shadow 0.2s;
+  &:last-of-type {
+    margin-bottom: 0;
   }
+  height: fit-content;
 
-  :after {
-    display: inline-block;
-    content: "";
-    width: 18px;
-    height: 18px;
-    background: ${palette.blue.header};
-    border-radius: 60px;
-    position: absolute;
-    top: 0;
-    left: 0;
-    bottom: 0;
-    margin: auto;
-    margin-left: 10px;
-    transition: 0.3s ease-in-out;
-    transition-duration: 0.2s;
-    transform: scale(${isSelected ? "0.6" : "0.2"});
-    opacity: ${isSelected ? "1" : "0"};
+  :hover {
+    input:not([disabled]) {
+      border-color: ${palette.neutral[100]};
+    }
   }
 `;
+const radioLabelTextStyles = (disabled: boolean) => css`
+  position: relative;
+  font-family: "Guardian Text Sans Web", Helvetica Neue, Helvetica, Arial,
+    Lucida Grande, sans-serif;
+  font-size: 17px;
+  color: ${disabled ? palette.brand.pastel : palette.neutral[100]};
+`;
 
 interface Props {
-  onChangeHandler: (value: boolean) => void;
-  selectedValue: ItemState;
+  onChangeHandler?: (value: boolean) => void;
+  selectedValue?: ItemState;
```

Comment:
Is it true that these are both optional? Would an empty object be a valid `Props`?

Reply:
Yes, both are optional. If one of them is missing we render the radio button as disabled.